system-addon/content-src/lib/init-store.js

```diff
@@ -58,6 +71,10 @@ const rehydrationMiddleware = store => next => action => {
 
   if (isMergeStoreAction) {
     store._didRehydrate = true;
+    if (store._earlyMessageQueue.length) {
+      store._earlyMessageQueue.forEach(a => store.dispatch(a));
```

Comment:
Not sure if there's any concern of potentially mixed actions? E.g., getting an action dispatched in the middle of another action (vs after -- but it's not uncommon for an action to trigger a subsequent action). I would guess maybe the new action is appended to some action queue that is dispatched after the current action… @k88hudson ?

Reply:
Just to be clear, when you say "dispatched *inside* its middleware", is this via `next` instead of `store.dispatch`?